node/src/p2p.rs

```diff
@@ -106,6 +123,18 @@ pub enum P2pError {
     /// Bootnode address is missing its peer ID.
     #[error("Bootnode multiaddrs without peer ID: {0:?}")]
     BootnodeAddrsWithoutPeerId(Vec<Multiaddr>),
+
+    /// An error propagated from [`beetswap::Behaviour`].
+    #[error("Bitswap: {0}")]
+    Bitswap(#[from] beetswap::Error),
+
+    /// ProtoBuf message failed to be decoded.
+    #[error("ProtoBuf decoding error: {0}")]
+    ProtoDecodeFailed(celestia_tendermint_proto::Error),
```

Comment:
any reason to not use `#[from]` there?

Reply:
and `Ok(foo::decode(...)?)` instead of map_err

Reply:
no reason. added it